tools/kliopt/org/jetbrains/kliopt/KliOpt.kt

```diff
@@ -173,8 +174,36 @@ class ArgParser(optionsList: List<OptionDescriptor>, argsList: List<ArgDescripto
         if (!descriptor.isMultiple && !processedValues.getValue(descriptor.longName).isEmpty()) {
             printError("Option ${descriptor.longName} is used more than one time!")
         }
-        descriptor.type.check(value, descriptor.longName)
-        processedValues.getValue(descriptor.longName).add(value)
+        val savedValues = descriptor.delimiter?.let {
+            // Escape values on quotes.
```

Comment:
What does this fix?

Reply:
This is fix for parameters in quotes with space. If parameter can be written with some delimiter, current parser should work properly with them after default compiler parser is fixed(issue discussed in slack)

Reply:
As far as I can see, this code can't fix the issue discussed.

Reply:
This code process case when user can write cinterop -lopt "-f \"some name\" -another_flag". String inside quotes wouldn't be splitted.

Reply:
May be it isn't solution for full problem, but as I understood such case also can be and should be parsed properly.

Reply:
> This code process case when user can write cinterop -lopt "-f "some name" -another_flag". String inside quotes wouldn't be splitted.

User is not supposed to pass such an option, because it doesn't have any effect. Linker options are passed when compiling the binary and handled with compiler arguments parser.

https://youtrack.jetbrains.com/issue/KT-29970
How does this patch fix anything mentioned in this issue?